config/passport.js

```diff
@@ -1,31 +1,60 @@
 import passport from 'passport';
-import bcrypt from 'bcrypt';
 import local from 'passport-local';
-import db from '../models';
+import GoogleStrategy from 'passport-google-oauth20';
+import FacebookStrategy from 'passport-facebook';
+import AuthController from '../controllers/authController';
+import User from '../models/User';
 
 const LocalStrategy = local.Strategy;
 
-passport.use(new LocalStrategy(
-  {
-    usernameField: 'user[email]',
-    passwordField: 'user[password]'
-  },
-  ((email, password, done) => {
-    db.User.findOne({ where: { email } })
-      .then((user) => {
-        bcrypt.compare(
-          password,
-          user.hashedPassword,
-          (err, res) => {
-            if (!user || !res || err) {
-              return done(null, false, {
-                errors: { 'email or password': 'is invalid' }
-              });
-            }
-            return done(null, user);
+const passportConfig = (app) => {
+  app.use(passport.initialize());
+  app.use(passport.session());
+  passport.use(new LocalStrategy(
+    {
+      usernameField: 'user[email]',
+      passwordField: 'user[password]'
+    },
+    ((email, password, done) => {
+      User.findOne({ email })
+        .then((user) => {
+          if (!user || !user.validPassword(password)) {
+            return done(null, false, {
+              errors: { 'email or password': 'is invalid' }
+            });
           }
-        );
-      })
-      .catch(done);
-  })
-));
+
+          return done(null, user);
+        })
+        .catch(done);
+    })
+  ));
+
+  /**
+ * configurations for passport strategies
+ */
+
+  passport.use(new GoogleStrategy({
+    clientID: process.env.GOOGLE_CLIENT_ID,
+    clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+    callbackURL: 'http://127.0.0.1:3000/api/users/login/google/redirect',
+  }, AuthController.callback));
+
+  passport.use(new FacebookStrategy({
+    clientID: process.env.FACEBOOK_APP_ID,
+    clientSecret: process.env.FACEBOOK_APP_SECRET,
+    callbackURL: 'http://localhost:3000/api/users/login/facebook/redirect',
```

Comment:
I'll suggest you make redirect URL  relative than absolute because when the app will be deployed, the domain will change.